### DISCUSSION TITLE
Differentiating between UaExceptions

### DISCUSSION THREAD

**Discussion by bsalmilo** [eclipse-milo/milo#1284] [General]
Hi, 

I'm trying to work on error handling in my project. One of the things I'm struggling with is is the difference between various UaExceptions, specifically whether the exception was thrown because of an error with the UaClient or an error on my system versus an error thrown by the server due to address space browsing. Is there any info you could give me on this, or are there any docs or resources you could point me to? I'm having trouble finding them searching on my own. 

Thank you

**Comment by kevinherron** [comment]
Unfortunately this is a shortcoming in the current API/design, there is no deterministic way to determine whether the error originated in the client or server. You have to examine the StatusCode and stack trace to make a decision about it.
